src/Consolonia.Core/Drawing/PixelBufferImplementation/EgaConsoleColor/EgaConsoleColorMode.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            (ConsoleColor consoleColor1, EgaColorMode mode1) = ConvertToConsoleColorMode(color1);
+            (ConsoleColor consoleColor2, EgaColorMode mode2) = ConvertToConsoleColorMode(color2);
+
+            switch (mode2)
+            {
+                case EgaColorMode.Transparent:
+                    return color1;
+
+                case EgaColorMode.Shaded when mode1 == EgaColorMode.Shaded:
+                {
+                    ConsoleColor doubleShadedColor = Shade(Shade(consoleColor1));
+                    return ConvertToAvaloniaColor(doubleShadedColor);
+                }
+                case EgaColorMode.Shaded:
+                {
+                    ConsoleColor shadedColor = Shade(consoleColor1);
+                    return ConvertToAvaloniaColor(shadedColor);
+                }
+                case EgaColorMode.Colored:
+                default:
+                    return ConvertToAvaloniaColor(consoleColor2);
+            }
+        }
+
+        public void SetAttributes(InputLessDefaultNetConsole console, Color background, Color foreground,
+            FontWeight? weight)
+        {
+            (ConsoleColor backgroundConsoleColor, EgaColorMode mode) = ConvertToConsoleColorMode(background);
+            if (mode is not EgaColorMode.Colored)
+                ConsoloniaPlatform.RaiseNotSupported(62144, foreground);
+
+
+            (ConsoleColor foregroundConsoleColor, _) = ConvertToConsoleColorMode(foreground);
+            //todo: if mode is transparent, don't print foreground. if shaded - shade it
+
+            var sb = new StringBuilder();
+
+            // Append ANSI escape sequence for background color
+            sb.Append(GetAnsiCode(backgroundConsoleColor, true));
+
+            // Append ANSI escape sequence for foreground color
+            sb.Append(GetAnsiCode(foregroundConsoleColor, false));
+            console.WriteText(sb.ToString());
+            return;
+
+            // Function to map ConsoleColor to ANSI code
+            static string GetAnsiCode(ConsoleColor color, bool isBackground)
+            {
+                int ansiCode = color switch
+                {
+                    ConsoleColor.Black => 0,
+                    ConsoleColor.DarkRed => 1,
+                    ConsoleColor.DarkGreen => 2,
+                    ConsoleColor.DarkYellow => 3,
+                    ConsoleColor.DarkBlue => 4,
+                    ConsoleColor.DarkMagenta => 5,
+                    ConsoleColor.DarkCyan => 6,
+                    ConsoleColor.Gray => 7,
+                    ConsoleColor.DarkGray => 8,
+                    ConsoleColor.Red => 9,
+                    ConsoleColor.Green => 10,
+                    ConsoleColor.Yellow => 11,
+                    ConsoleColor.Blue => 12,
+                    ConsoleColor.Magenta => 13,
+                    ConsoleColor.Cyan => 14,
+                    ConsoleColor.White => 15,
+                    _ => 7 // Default to white if unknown
```

Comment:
is it possible to detect default color when app starts?  Maybe future enhancement.